### PR TITLE
DAPI-154 - Add staff endpoint

### PR DESCRIPTION
**Pull request by electrified**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/dto/v1/staff/NewStaff.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  @field:Size(max = 35)
+  val forename: String,
+
+  @ApiModelProperty("The provider code of the staff member")
+  @field:NotBlank
+  @ProviderCode
+  val provider: String,
+
+  val teams: List<String>?,
+)
```
Comment (axle-h) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/dto/v1/staff/NewStaff.kt:25]: These field names conflict with the NSI PR, I used more sensible names.
image
What do you think?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/dto/v1/staff/StaffDto.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.justice.digital.hmpps.deliusapi.dto.v1.staff
+
+import java.time.LocalDate
+
+data class StaffDto(
+  val code: String,
+  var startDate: LocalDate,
+  val surname: String,
+  val forename: String,
+  val private: Boolean,
```
Comment (axle-h) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/dto/v1/staff/StaffDto.kt:10]: Does this not cause any issues?! Would prefer not to use reserved words where possible.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/dto/v1/staff/NewStaff.kt (hunk 1 above)
Comment (electrified) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/dto/v1/staff/NewStaff.kt:25]: Sure, happy to go with your naming

File: src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/dto/v1/staff/StaffDto.kt (hunk 2 above)
Comment (electrified) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/dto/v1/staff/StaffDto.kt:10]: Haha, it works fine. IntelliJ's syntax highlighting is a bit more intelligent than githubs and doesn't colour it differently. I can change it though.

image

Comment (axle-h) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/dto/v1/staff/StaffDto.kt:10]: Ah nice, nah if there's no issues using this name then I'm sure it'll be fine. I suppose we'd also want to ensure our clients can handle it ok too. Does Java in community-api have issues with a field named 'private' ?

Comment (axle-h) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/dto/v1/staff/StaffDto.kt:10]: Tyepscript can handle it: https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgCIFcC2mCeyDeAUMicgA5TABuckAXMgMpiUgDmA3MaQPZgAW0ACr9QbBs1adCAX0KEEPEAGcwyAB4MM2PAF4C5SjXrIA5BWq0IpgDTI+gqCLENTD6AFoBY08hldFFR4AGwgAOmCeNgAKdQBKDiA

**Pull request merged**